.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,24 @@
+# https://github.com/pre-commit/action
```

Comment:
Would you mind sharing that code under the MIT license and adding an MIT license header here? Mimicking the format at… https://github.com/hahey/django-migration-vis/blob/c97016d764571cad269ab0ed3a8c926dfed25256/django_migration_vis/management/commands/visualizemigrations.py#L1-L3 …would be perfect.

Reply:
That project is not mine and i don't intend to maintain it with a different license.

Reply:
I think we may have a misunderstanding. Let's try to clear it up. The project is licensed under MIT right now. It is good practice to have a copyright header mentioning the license in all files possible. And by adding that line you'll confirm to the community (not to keep maintaining this file or project but) that you yourself had the right to contribute this patch (e.g. some work contracts would take away those rights from you and then you cannot contribute anywhere) and agree to the MIT license. What do you think?

Reply:
Oh okay got it, thanks for the clarification.

Reply:
@hartwork On second thought, and again I'm okay with whatever you see fit but I'm just sharing my opinion, I believe license information belongs to the root directory of repo, not as duplicated lines in each file (and it creates clutter) Github detect top-level LICENSE files and assign them to repo attributes.

Reply:
@ulgens the problem starts e.g. when people want to copy files to other projects, when you have mixed license code bases and so on. GitHub's detection cannot be relied on, they only show one single license and do not even distinct between GPL-2 and GPL-2+ properly. Do they by now? I have seen situations where files without explicit headers were a problem and the FSF also explicitly recommends it:

> Give each file the proper copyright notices. Make sure to clearly identify which versions of the license users can use.

From https://www.gnu.org/licenses/gpl-howto.en.html . I see no reason to limit that to GPL.

Reply:
Fixed.